### DISCUSSION TITLE
objectbuckets.objectbucket.io stuck in deletion.

### DISCUSSION THREAD

**Discussion by manjo-git** [rook/rook#9414] [General]
I had a ceph deployed with s3 buckets, objectbuckets.objectbucket.io is stuck in deletion when I tried to tear down the cluster. I tried removing crd finalizers but when I try to fush the disks and recreate the cluster it says 

`Error from server (AlreadyExists): error when creating "crds.yaml": object is being deleted: customresourcedefinitions.apiextensions.k8s.io "objectbuckets.objectbucket.io" already exists`

How do I manually delete the bucket and objectbuckets.objectbucket.io

 tried this but does not work.

```
>  kubectl -n rook-ceph patch objectbuckets.objectbucket.io --type merge -p '{"metadata":{"finalizers": [null]}}'
error: resource(s) were provided, but no name was specified
```

**Comment by travisn** [comment]
Did you also delete the CephCluster CR? If both the cluster and object store CRs are deleted, the operator should remove the finalizers for you. But perhaps if you deleted the operator too soon, it could not remove the finalizers for you. Did you remove the finalizer on the CephCluster CR as well?

**Reply by manjo-git** [reply]
I moved the CR using: 

`kubectl -n rook-ceph patch cephcluster rook-ceph --type merge -p '{"spec":{"cleanupPolicy":{"confirmation":"yes-really-destroy-data"}}}'
`

I also did : 

```
for CRD in $(kubectl get crd -n rook-ceph | awk '/ceph.rook.io/ {print $1}'); do
    kubectl get -n rook-ceph "$CRD" -o name | \
    xargs -I {} kubectl patch -n rook-ceph {} --type merge -p '{"metadata":{"finalizers": [null]}}'
done
```

and also

```
kubectl api-resources --verbs=list --namespaced -o name \
  | xargs -n 1 kubectl get --show-kind --ignore-not-found -n rook-ceph
```

and removed the configmap and secrets. It did list the bucket in the config map secrets. Also, I went to each node deleted the /var/lib/rook/* and /var/lib/kubelet/plugins/rook-ceph.*

when I create the cluster kubectl create -f crds.yamls fails with 

`Error from server (AlreadyExists): error when creating "crds.yaml": object is being deleted: customresourcedefinitions.apiextensions.k8s.io "objectbuckets.objectbucket.io" already exists
`

I think I need a way to remove "objectbuckets.objectbucket.io"

**Reply by travisn** [reply]
@manjo-git With this error I would expect that CRs of that type still exist. Do you see any of these resources with `kubectl get objectbuckets`? If so you would need to remove those, and then the CRD should be allowed to be deleted.

**Reply by manjo-git** [reply]
yes you are right .. it still exists.

```
> kubectl get objectbuckets
NAME                               AGE
obc-rook-ceph-ceph-delete-bucket   3d7h
```
How do I delete it ? delete seems to be stuck for ever.

```
> kubectl delete objectbucket obc-rook-ceph-ceph-delete-bucket
objectbucket.objectbucket.io "obc-rook-ceph-ceph-delete-bucket" deleted

```

**Reply by manjo-git** [reply]
Looks like I was able to delete it as follows: 

```
>  kubectl patch   objectbucket obc-rook-ceph-ceph-delete-bucket  --type merge -p '{"metadata":{"finalizers": [null]}}'
objectbucket.objectbucket.io/obc-rook-ceph-ceph-delete-bucket patched

>   kubectl delete objectbucket obc-rook-ceph-ceph-delete-bucket
error: the server doesn't have a resource type "objectbucket"

> kubectl get objectbucket
error: the server doesn't have a resource type "objectbucket"
```

**Reply by manjo-git** [reply]
thanks a ton @travisn I was able to re-deploy crds.yaml again.
